manual backport of docs from commit abac619 into release/1.17.x (#31194)
hashicorp/vault: website/content/docs/commands/plugin/register.mdx

The `plugin register` command registers a new plugin in Vault's plugin catalog.
The plugin's type of "auth", "database", or "secret" must be included.

<Note>

> To register an Enterprise plugin, you must register with the plugin artifact, not the binary.

Reviewer: ```suggestion
To register an enterprise plugin, you must register with the extracted
`.zip` file, not the binary.
```
Technically "artifact" could refer to anything. Even a binary file can be an artifact in some cases. Since we mean "extracted .zip file", it's clearer for readers if we actually say that

Reviewer: It's up to you if you want to apply the full suggestion or just the capitalization change. I'm realizing I probably missed the "artifact" usage in the original PR, so feel free to ignore that part so the docs stay consistent. We/I can go back to tweak the language later

    auth my-custom-plugin
```

> Register a plugin artifact with custom args:

Reviewer: ```suggestion
Register a plugin `.zip` file with custom args:
```

Reviewer: Same note re: ignoring this suggestion

Reviewer: sgtm

### Command options

- `sha256` `(string: "")` – The SHA256 sum of a plugin binary or the OCI image.
  You must provide `sha256` to register a plugin binary, but you must leave
  `sha256` unset to register a plugin artifact.

- `-args` `([]string: [])` - Argument to pass to the plugin when starting. This
  flag can be specified multiple times to specify multiple args.

- `-command` `(string: "")` - Specifies the command path used to execute the
  plugin relative to the plugin directory or OCI image directory. You must
  provide `command` to register a plugin binary. Vault ignores `command` when

>   you register a plugin artifact as it already knows the associated run command.

Reviewer: ```suggestion
  you register with an extracted `.zip` file as it already knows the associated run command.
```

Reviewer: Same note re: ignoring this suggestion
